credentials/generate-revocation-set.py

```diff
@@ -491,13 +591,12 @@ def main(use_main_net_dcld: str, use_test_net_dcld: str, use_main_net_http: bool
         # 10. Iterate through the Revoked Certificates List
         for revoked_cert in crl_file:
             try:
-                revoked_cert_issuer = revoked_cert.extensions.get_extension_for_oid(
-                    x509.CRLEntryExtensionOID.CERTIFICATE_ISSUER).value.get_values_for_type(x509.DirectoryName).value
-
+                revoked_cert_issuer = base64.b64encode(revoked_cert.extensions.get_extension_for_oid(
```

Comment:
In spirit similar to my previous comments, I think we should decode as much as we can and use the result to pass around, rather than using b64. b64 is good when you need to pass printable chars for binary data, but when passing things in/out of the function, we should try using the fully decoded value which is essentially a string. Also, this way, during debugging, one can easily inspect the value without having to decode, (and in this case) needing to encode just to compare.

Reply:
I've updated the handling and the comparison done in the final step. Now we are not passing around the b64 encoded name. After some cleanup the x509.name.Name object is passed and its helper functions are used to get human readable names, and in the get_approved_certificate function the b64 conversion happens as its needed for the request. At the end the output now provides the human readable name.

Reply:
Cool. Thanks.